### PR TITLE
V2: go serverless!

### PR DESCRIPTION
**Pull request by amio**
## Goals

- Serverless oriented (the reason for v2)
- Rewrite homepage in React (better expressiveness)
- Rewrite core in TypeScript

## Changes

- Sunset `api.badgen.net`. It's not a must have, and we might easily exceeds rate limit (on a live badge using private token for api requesting) by serving this.
- Reveal badge builder https://v2.badgen.net/builder
- Auto generate doc page for all badges.
- `runkit` and `https` docs overhauled.
- `lgtm` badge upgraded with new api. More badges.

## Todos

- [x] Serve flat style
- [x] Migrate all live badges
- [x] Serve docs at `/docs/<badge-name>`
- [x] Error proof

## Preview

https://v2.badgen.net

### CODE REVIEW

**Comment by amio** [comment]
This pull request **introduces 2 alerts** when merging b3566e4ee768ae1f0b7ccea35dc8028bc7b2dbb8 into 949b65f83872d3c13d8e93916ed11e3213f1e28e - [view on LGTM.com](https://lgtm.com/projects/g/badgen/badgen.net/rev/pr-1a169fcfb429da07d994f47a19a520abdfcb5fa7)

**new alerts:**

* 1 for Useless conditional
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by amio** [comment]
This pull request **introduces 2 alerts** when merging f3341b80de2f32ae00cbb11f652517d8cab2172b into 949b65f83872d3c13d8e93916ed11e3213f1e28e - [view on LGTM.com](https://lgtm.com/projects/g/badgen/badgen.net/rev/pr-d3bfcd667cddf99969833390ac8cf42a4c7a8115)

**new alerts:**

* 1 for Useless conditional
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "noImplicitAny": false,
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "skipLibCheck": true,
+    "jsx": "preserve",
+    "target": "es2017",
```
Comment (styfle) [inline tsconfig.json]:
```suggestion
    "target": "esnext",
```

Probably don't need any polyfills so I usually just keep this as esnext when targeting Node.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "noImplicitAny": false,
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
```
Comment (styfle) [inline tsconfig.json]: I don't think you need `allowSyntheticDefaultImports` when `esModuleInterop` is true

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "compilerOptions": {
+    "strict": true,
+    "noImplicitAny": false,
```
Comment (styfle) [inline tsconfig.json]: I don't think you need `noImplicitAny` when `strict` is true.

File: tsconfig.json (hunk 1 above)
Comment (amio) [reply tsconfig.json]: Yep, it should be `esnext`.

BTW, it confused me that VSCode keep warning about `Object.entries()`, while compiling with tsc is fine. I tried various ways to fix it, none of them works. So I end up use `// @ts-ignore` for this https://github.com/badgen/badgen.net/pull/282/files#diff-577c40df72d7574b39845c9931c83104R5 😕

File: tsconfig.json (hunk 2 above)
Comment (amio) [reply tsconfig.json]: Great, thanks!

**Comment by amio** [comment]
This pull request **introduces 2 alerts** when merging 1fbd749beb060c9074703da48e3030590bf3599b into 949b65f83872d3c13d8e93916ed11e3213f1e28e - [view on LGTM.com](https://lgtm.com/projects/g/badgen/badgen.net/rev/pr-3baed7bb86d50c401eefc5268694721f76f24b48)

**new alerts:**

* 1 for Useless conditional
* 1 for Unused variable, import, function or class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: tsconfig.json (hunk 3 above)
Comment (amio) [reply tsconfig.json]: It's meant to disable it ;)

**Comment by amio** [comment]
https://flat.badgen.net is on v2 now 🥳

**Comment by amio** [comment]
After switching https://flat.badgen.net to v2, the "Serverless Execution" fee goes up to ~$2 a day, that is $60 a month. Compare to $14.99 a month (for the whole badgen.net) previously, I think we should stick to v1. Might revisit on moving to v2 after zeit/now-cli#2399 gets resolved.

image

**Pull request merged**